processor/src/debug.rs

```diff
@@ -132,11 +166,13 @@ impl Iterator for VmStateIterator {
         if is_start {
             self.asmop_idx += 1;
         }
+        self.set_proc_stack();
```

Comment:
I think trying to update `proc_stack` here creates a lot of complications (e.g., you need to keep track of procedures to pop at the next cycle). Maybe you can do a part here and then a part after the result has been built (i.e., after line 182)? This way, `proc_stack` will always be in the correct state.